walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -75,22 +75,21 @@
       TimeRange proposedSlot = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, times.get(0).start(), false);
       addIfProposedTimeIsLongEnough(proposedSlot, request, complementTimes);
     }
-    int i;
-    int next;
-    for (i = 0, next = i+1; next < times.size(); i++, next++) {
+    int i = 0;
+    int next = 1;
+    while (next < times.size()) {
       // If there is any space between the current time and the next time, add it to the list
       // Otherwise, keep current time the same until the "next" event is one not entirely contained in the current one 
       if (!times.get(i).overlaps(times.get(next))) {
         TimeRange proposedSlot = TimeRange.fromStartEnd(times.get(i).end(), times.get(next).start(), false);
         addIfProposedTimeIsLongEnough(proposedSlot, request, complementTimes);
       }
-    	if (times.get(i).contains(times.get(next))) {
-        i--;
-        continue;
-      } else {
+    	if (! times.get(i).contains(times.get(next))) {
+        i++;
         // Reset next to be event directly following current one, if next is not nested within the current event
         next = i+1;
```

Comment:
do you also want the next++ after this statement in this case?

Reply:
I think so. The code is still passing all tests. The `next++` was previously in the update statement and wouldn't have been skipped over by the `continue` keyword either way.